pkg/reconciler/v1alpha1/revision/resources/env_var.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/logging"
+	"github.com/knative/serving/pkg/reconciler/v1alpha1/revision/resources/names"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	knativeEnvVariableKey = "KNATIVE_APPLICATION"
```

Comment:
The concept of "application" is not really defined in Knative. What about renaming the env var to `KNATIVE_CONTEXT` ?

Reply:
We debated between one variable or multiple, but ultimately went with one variable because we felt it gave us a little bit more flexibly in the future with adding more variables with less impact to the app/function. We also had experience with using a similar approach with `Cloud Foundry` and `Spring` and both use the one field with json